src/components/complibrary/buildproductcardfavorites.jsx

```diff
@@ -0,0 +1,88 @@
+import React, {Component} from 'react';
+import {CardText,Card,CardTitle, CardMenu, IconButton} from 'react-mdl';
+import {Link} from 'react-router-dom';
+
+class BuildProductCardFavorites extends Component {
+
+    constructor(props){
+        super(props);
+        this.state = {
+            //empty for now
+        }
+    }
+    
+
+    componentDidMount(){
+    }
+
+    render() {
+        const product = this.props.productFromParent;
+        var productNameEllipsis = product.productName.toString().substring(0,29) + "...";
```

Comment:
`defaultProps` is helpful for safety in these components
maybe you can reuse this code and https://github.com/reetom/smartcomm/pull/1/files#diff-bb0328b4cf2c208889bcda89536e7bb7R23 this code as a function like `fromProductToProductName`?